android/src/main/java/io/invertase/firebase/notifications/RNFirebaseBackgroundNotificationReceiver.java

```diff
@@ -0,0 +1,47 @@
+package io.invertase.firebase.notifications;
+
+import android.app.ActivityManager;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.util.Log;
+
+import com.facebook.react.HeadlessJsTaskService;
+
+import java.util.List;
+
+public class RNFirebaseBackgroundNotificationReceiver extends BroadcastReceiver {
+
+  @Override
+  public void onReceive(Context context, Intent intent) {
+    Intent serviceIntent = new Intent(context, RNFirebaseBackgroundNotificationsService.class);
+    serviceIntent.putExtras(intent.getExtras());
+    context.startService(serviceIntent);
+
+    if (!isAppOnForeground((context))) {
+      HeadlessJsTaskService.acquireWakeLockNow(context);
+    }
+  }
+
+  private boolean isAppOnForeground(Context context) {
```

Comment:
Can you change this to use the method from the `io.invertase.firebase.Utils` class rather than duplicating?

Reply:
Done